core-tests/shared/src/test/scala/zio/duration/DurationSpec.scala

```diff
@@ -3,416 +3,320 @@ package zio.duration
-
-  def nano7L =
-    1L.microsecond must_=== Duration.fromNanos(1000L)
-
-  def nano8L =
-    2L.microseconds must_=== Duration.fromNanos(2000L)
-
-  def nano9L =
-    1L.milli must_=== Duration.fromNanos(1000000L)
-
-  def nano10L =
-    2L.millis must_=== Duration.fromNanos(2000000L)
-
-  def nano11L =
-    1L.millisecond must_=== Duration.fromNanos(1000000L)
-
-  def nano12L =
-    2L.milliseconds must_=== Duration.fromNanos(2000000L)
-
-  def nano13L =
-    1L.second must_=== Duration.fromNanos(1000000000L)
-
-  def nano14L =
-    2L.seconds must_=== Duration.fromNanos(2000000000L)
-
-  def nano15L =
-    1L.minute must_=== Duration.fromNanos(60000000000L)
-
-  def nano16L =
-    2L.minutes must_=== Duration.fromNanos(120000000000L)
-
-  def nano17L =
-    1L.hour must_=== Duration.fromNanos(3600000000000L)
-
-  def nano18L =
-    2L.hours must_=== Duration.fromNanos(7200000000000L)
-
-  def nano19L =
-    1L.day must_=== Duration.fromNanos(86400000000000L)
-
-  def nano20L =
-    2L.days must_=== Duration.fromNanos(172800000000000L)
-
-  def nano1 =
-    1.nano must_=== Duration.fromNanos(1L)
-
-  def nano2 =
-    2.nanos must_=== Duration.fromNanos(2L)
-
-  def nano3 =
-    1.nanosecond must_=== Duration.fromNanos(1L)
-
-  def nano4 =
-    2.nanos must_=== Duration.fromNanos(2L)
-
-  def nano5 =
-    1.micro must_=== Duration.fromNanos(1000L)
-
-  def nano6 =
-    2.micros must_=== Duration.fromNanos(2000L)
-
-  def nano7 =
-    1.microsecond must_=== Duration.fromNanos(1000L)
-
-  def nano8 =
-    2.microseconds must_=== Duration.fromNanos(2000L)
-
-  def nano9 =
-    1.milli must_=== Duration.fromNanos(1000000L)
-
-  def nano10 =
-    2.millis must_=== Duration.fromNanos(2000000L)
-
-  def nano11 =
-    1.millisecond must_=== Duration.fromNanos(1000000L)
-
-  def nano12 =
-    2.milliseconds must_=== Duration.fromNanos(2000000L)
-
-  def nano13 =
-    1.second must_=== Duration.fromNanos(1000000000L)
-
-  def nano14 =
-    2.seconds must_=== Duration.fromNanos(2000000000L)
-
-  def nano15 =
-    1.minute must_=== Duration.fromNanos(60000000000L)
-
-  def nano16 =
-    2.minutes must_=== Duration.fromNanos(120000000000L)
-
-  def nano17 =
-    1.hour must_=== Duration.fromNanos(3600000000000L)
-
-  def nano18 =
-    2.hours must_=== Duration.fromNanos(7200000000000L)
-
-  def nano19 =
-    1.day must_=== Duration.fromNanos(86400000000000L)
-
-  def nano20 =
-    2.days must_=== Duration.fromNanos(172800000000000L)
-
-}
+object DurationSpec
+    extends ZIOBaseSpec(
+      suite("DurationSpec")(
+        test("Make a Duration from positive nanos and check that: The Duration is Finite") {
+          assert(Duration.fromNanos(1), isSubtype[Duration.Finite](Assertion.anything))
+        },
+        test("Make a Duration from positive nanos and check that: Copy with a negative nanos returns Zero") {
+          assert(Duration.fromNanos(1).asInstanceOf[Duration.Finite].copy(-1), isSubtype(equalTo(Duration.Zero)))
```

Comment:
I don't think you need the `isSubtype` here since you are already checking for equality.

Reply:
Thanks, good catch, I tried to do it without asInstanceOf and this is an old artefact from that.